apps/dashboard/app/models/remote_file.rb

```diff
@@ -77,8 +77,15 @@ def write(content)
   end
 
   def handle_upload(tempfile)
-    # FIXME: upload to the remote asynchronously
-    RcloneUtil.moveto(remote, path, tempfile.path)
+    # Start a transfer that moves the Rack tempfile to the remote.
+    transfer = RemoteTransfer.build(
+      action: "mv",
+      files: { tempfile.path => path.to_s },
+      src_remote: RcloneUtil::LOCAL_FS_NAME,
+      dest_remote: remote,
+      tempfile: tempfile
+    )
+    transfer.tap(&:perform_later)
```

Comment:
digging through `‎RcloneUtil.rclone_with_progress` I wonder what happens when `rclone moveto` fails? I'm guessing Rack has some way to delete the file on failure. Do we have such a safety mechanism?

Reply:
`tempfile` here is a [Tempfile](https://ruby-doc.org/stdlib-3.1.1/libdoc/tempfile/rdoc/Tempfile.html), so it is cleaned up on garbage collection (which is why I pass the whole thing to RemoteTransfer) or when the Ruby interpreter exits.
I pushed a commit to explicitly clean up (close) the file on transfer errors, so now it should always be cleaned up instantly either when the transfer completes or fails.